test/run.d

```diff
@@ -147,26 +151,53 @@ void verifyCompilerExists(string[string] env)
 Builds the binary of the tools required by the testsuite.
 Does nothing if the tools already exist and are newer than their source.
 */
-void ensureToolsExists(const TestTool[] tools ...)
+void ensureToolsExists(string[string] env, const TestTool[] tools ...)
 {
     resultsDir.mkdirRecurse;
 
     foreach (tool; tools.parallel(1))
     {
-        const targetBin = resultsDir.buildPath(tool).exeName;
-        const sourceFile = toolsDir.buildPath(tool ~ ".d");
+        string targetBin;
+        string sourceFile;
+        if (tool.linksWithTests)
```

Comment:
Maybe add a one-sentence here explaining sth. like this is preparing the shell tester D library  and thus needs the actual DMD compiler under test?

Reply:
I added a sentence where I add the compiler with `env["DMD"]`